src/v3/src/components/PasswordEnrollment/PasswordEnrollment.tsx

```diff
@@ -0,0 +1,117 @@
+
+import { h } from 'preact';
+import { useState } from 'preact/hooks';
+
+import { useOnSubmit, useValue } from '../../hooks';
+import { useTranslation } from '../../lib/okta-i18n';
+import {
+  ChangeEvent,
+  PasswordEnrollmentElement,
+  SubmitEvent,
+  UISchemaElementComponent,
+} from '../../types';
+import InputPassword from '../InputPassword';
+
+const PasswordEnrollment: UISchemaElementComponent<{
+  uischema: PasswordEnrollmentElement
+}> = ({ uischema }) => {
+  const { ctaLabel, input } = uischema.options;
+
+  const { t } = useTranslation();
+  const value = useValue(input);
+  const [isTouched, setIsTouched] = useState<boolean>(false);
+  const [confirmPassword, setConfirmPassword] = useState<string>();
+  const [confirmPasswordError, setConfirmPasswordError] = useState<string | undefined>();
+  const onSubmitHandler = useOnSubmit();
+
+  const handleConfirmPasswordValidation = (password: string | undefined): void => {
+    if (!password) {
+      setConfirmPasswordError(t('model.validation.field.blank'));
+      return;
+    }
+
+    if (value !== password) {
+      setConfirmPasswordError(t('password.error.match'));
+      return;
+    }
+
+    setConfirmPasswordError(undefined);
+  };
+
+  const handleSubmit = async (e: SubmitEvent) => {
+    e.preventDefault();
+    if (confirmPassword === value) {
+      onSubmitHandler({
+        includeData: true,
+      });
+      return;
+    }
+    setIsTouched(true);
+    handleConfirmPasswordValidation(confirmPassword);
+  };
+
+  const handleConfirmPasswordChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setIsTouched(true);
+    setConfirmPassword(e.currentTarget.value);
+    handleConfirmPasswordValidation(e.currentTarget.value);
+  };
+
+  const handleConfirmPasswordBlur = () => {
+    setIsTouched(true);
+    handleConfirmPasswordValidation(confirmPassword);
+  };
+
+  return (
+    <Box>
+      <Box marginBottom={4}>
+        <InputPassword uischema={input} />
+      </Box>
+      <Box marginBottom={4}>
+        <PasswordInput
+          label={t('oie.password.confirmPasswordLabel')}
+          value={confirmPassword}
+          name="credentials.confirmPassword"
+          id="credentials.confirmPassword"
+          error={!!(isTouched && confirmPasswordError)}
+          helperText={confirmPasswordError}
+          onBlur={handleConfirmPasswordBlur}
+          onChange={handleConfirmPasswordChange}
+          fullWidth
+          inputProps={{
+            'data-se': 'credentials.confirmPassword',
+            autocomplete: 'new-password',
+          }}
+        />
+        {confirmPasswordError && (
+          <FormHelperText error>{confirmPasswordError}</FormHelperText>
+        )}
+      </Box>
+      <Button
+        type="submit"
+        variant="primary"
+        fullWidth
+        data-se="#/properties/submit"
+        data-type="save"
+        onClick={handleSubmit}
+      >
+        {t(ctaLabel)}
+      </Button>
```

Comment:
Add note here that this handles submit button.

Reply:
see [here](https://github.com/okta/okta-signin-widget/pull/2632/commits/b2ce5b2cd2f7d7555b34c990642357150d2e8cc9), you can update dataSchemaRef in context with a new validate function to exclude submit button in this component.

Reply:
Ahh, this is nifty and would work perfectly for me. Will wait until OV enrollment fix is merged so I can pick up these changes and apply them.